test/master-playlist-controller.test.js

```diff
@@ -1283,6 +1283,31 @@ QUnit.test('selects a playlist after main/combined segment downloads', function(
   assert.equal(this.player.tech_.hls.stats.bandwidth, 4194304, 'default bandwidth');
 });
 
+QUnit.test('does not select a playlist after segment downloads if only one playlist', function(assert) {
+  let calls = 0;
+
+  this.masterPlaylistController.selectPlaylist = () => {
+    calls++;
+    return null;
+  };
+  this.masterPlaylistController.mediaSource.trigger('sourceopen');
+
+  // master
+  this.standardXHRResponse(this.requests.shift());
+  // media
+  this.standardXHRResponse(this.requests.shift());
```

Comment:
Is there a different initial playlist selection going on? I'd imagine the overriden `selectPlaylist` would return `null` for the media.

Reply:
So, it's loading the media manifest. I think what's happening is that since selectPlaylist is returning null, we don't end up switching to the media that's provided by selectPlaylist and we continue with the current select item.
Looking at the test above `selects a playlist after main/combined segment downloads`, changing it to return `playlist[1]` instead of `playlist[0]`, when inspecting this request, we get show it loading media1.m3u8, but in this case we still end up loading media.m3u8, which I believe is the expected behavior.

Reply:
Yeah, forgot about this line: https://github.com/videojs/http-streaming/blob/1.x/src/playlist-loader.js#L577